openfl-tutorials/interactive_api/HPU/PyTorch_MedMNIST_2D/workspace/HPU_Pytorch_MedMNIST_2D.ipynb

```diff
@@ -0,0 +1,622 @@
+    "\n",
+    "        self.layer3 = nn.Sequential(\n",
+    "            nn.Conv2d(16, 64, kernel_size=3),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU())\n",
+    "        \n",
+    "        self.layer4 = nn.Sequential(\n",
+    "            nn.Conv2d(64, 64, kernel_size=3),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU())\n",
+    "\n",
+    "        self.layer5 = nn.Sequential(\n",
+    "            nn.Conv2d(64, 64, kernel_size=3, padding=1),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU(),\n",
+    "            nn.MaxPool2d(kernel_size=2, stride=2))\n",
+    "\n",
+    "        self.fc = nn.Sequential(\n",
+    "            nn.Linear(64 * 4 * 4, 128),\n",
+    "            nn.ReLU(),\n",
+    "            nn.Linear(128, 128),\n",
+    "            nn.ReLU(),\n",
+    "            nn.Linear(128, num_classes))\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = self.layer1(x)\n",
+    "        x = self.layer2(x)\n",
+    "        x = self.layer3(x)\n",
+    "        x = self.layer4(x)\n",
+    "        x = self.layer5(x)\n",
+    "        x = x.view(x.size(0), -1)\n",
+    "        x = self.fc(x)\n",
+    "        return x\n",
+    "\n",
+    "# NOTE: This is a workaround to move model to HPU before initializing any optimizers \n",
+    "device = torch.device('hpu')\n",
+    "model = Net(in_channels=n_channels, num_classes=n_classes).to(device)\n",
+    "    \n",
+    "# define loss function and optimizer\n",
+    "if task == \"multi-label, binary-class\":\n",
+    "    criterion = nn.BCEWithLogitsLoss()\n",
+    "else:\n",
+    "    criterion = nn.CrossEntropyLoss()\n",
+    "    \n",
+    "optimizer = optim.SGD(model.parameters(), lr=lr, momentum=0.9)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f2154486",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d1c78ee",
+   "metadata": {},
+   "source": [
+    "### Register model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "59831bcd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from copy import deepcopy\n",
+    "\n",
+    "framework_adapter = 'openfl.plugins.frameworks_adapters.pytorch_adapter.FrameworkAdapterPlugin'\n",
+    "MI = ModelInterface(model=model, optimizer=optimizer, framework_plugin=framework_adapter)\n",
+    "\n",
+    "# Save the initial model state\n",
+    "initial_model = deepcopy(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "849c165b",
+   "metadata": {},
+   "source": [
+    "## Define and register FL tasks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ff463bd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TI = TaskInterface()\n",
+    "\n",
+    "train_custom_params={'criterion':criterion,'task':task}\n",
```

Comment:
Please do not pass complex Python objects or variables here. There is not logic on Collaborator's side that will pass `'multi-label, binary-class'` as task variable

Reply:
Got it, done.